api/proxy/common/eigenda_network.go

```diff
@@ -107,13 +115,36 @@ func EigenDANetworksFromChainID(chainID string) ([]EigenDANetwork, error) {
 	return networks, nil
 }
 
+// Useful when an rpc url is provided, but no network is specified.
+// In this case, we can use this function to automatically choose a default network.
+func DefaultEigenDANetworkFromChainID(chainID string) (EigenDANetwork, error) {
+	networks, err := EigenDANetworksFromChainID(chainID)
+	if err != nil {
+		return "", err
+	}
+	if len(networks) == 0 {
+		return "", fmt.Errorf("no EigenDA network found for chain ID: %s", chainID)
+	}
+	return networks[0], nil
+}
+
+// EigenDANetworkFromString parses an inputString to an EigenDANetwork value.
+// The returned EigenDANetwork is guaranteed to be non-nil.
+// If an invalid network is provided, an error is returned.
 func EigenDANetworkFromString(inputString string) (EigenDANetwork, error) {
 	network := EigenDANetwork(inputString)
 
 	switch network {
 	case HoleskyTestnetEigenDANetwork, HoleskyPreprodEigenDANetwork, SepoliaTestnetEigenDANetwork, MainnetEigenDANetwork:
 		return network, nil
 	default:
-		return "", fmt.Errorf("unknown network type: %s", inputString)
+		allowedNetworks := []string{
+			MainnetEigenDANetwork.String(),
```

Comment:
MainnetBeta missing

Reply:
Good point. Will add in a future PR if you don't mind, since I already need to refactor how networks are used as per austin's comment. Created https://linear.app/eigenlabs/issue/DAINT-662/make-eigenda-networks-statically-typed